### PR TITLE
feat/enterpriseportal: define GetCodyGatewayUsage RPC, use uint64 for rate limits

### PR DESCRIPTION
**Pull request by bobheadxi**
Part of https://linear.app/sourcegraph/issue/CORE-201. The RPC will be backed by https://github.com/sourcegraph/sourcegraph/pull/63528

The `uint64` for ratelimits is a backwards-compatible change, per proto docs:

> int32, uint32, int64, uint64, and bool are all compatible – this means you can change a field from one of these types to another without breaking forwards- or backwards-compatibility.

## Test plan

CI

### CODE REVIEW

**Comment by bobheadxi** [comment]
* **#63555** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/63555?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#63531** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/63531?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @bobheadxi and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by eseliger — APPROVED** [review]
kool

**Review by jac — APPROVED** [review]

File: lib/enterpriseportal/codyaccess/v1/codyaccess.proto [hunk 1]
```diff
@@ -27,6 +28,12 @@ service CodyAccessService {
   rpc UpdateCodyGatewayAccess(UpdateCodyGatewayAccessRequest) returns (UpdateCodyGatewayAccessResponse) {
     option idempotency_level = IDEMPOTENT;
   }
+
+  // GetCodyGatewayUsage returns data about a subscription's recent usage of
+  // Cody Gateway.
+  rpc GetCodyGatewayUsage(GetCodyGatewayUsageRequest) returns (GetCodyGatewayUsageResponse) {
+    option idempotency_level = IDEMPOTENT;
```
[the remark this answers is not in the document]
Comment (bobheadxi) [reply lib/enterpriseportal/codyaccess/v1/codyaccess.proto]: Thanks, fixed!

**Comment by bobheadxi** [comment]
### Merge activity

* **Jul 2, 12:25 PM EDT**: @bobheadxi merged this pull request with [Graphite](https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/63531).

**Pull request merged**